manager.go

```diff
@@ -0,0 +1,248 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the LGPL, see LICENCE file for details.
+
+package aclstore
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/julienschmidt/httprouter"
+	"gopkg.in/errgo.v1"
+	httprequest "gopkg.in/httprequest.v1"
+
+	"github.com/juju/aclstore/params"
+)
+
+// Params holds the parameters for a NewManager call.
+type Params struct {
+	// Store holds the persistent storage used by the handler.
+	Store ACLStore
+
+	// RootPath holds the root URL path prefix to use
+	// for the ACL endpoints. All the endpoints will be
+	// prefixed with this path.
+	RootPath string
+
+	// Authenticate authenticates the given HTTP request and returns
+	// the resulting authenticated identity. If authentication
+	// fails, Authenticate is responsible for writing its own
+	// response - the returned error can be used for logging.
+	Authenticate func(ctx context.Context, w http.ResponseWriter, req *http.Request) (Identity, error)
+
+	// InitialAdminUsers holds the contents of the admin ACL
+	// when it is first created.
+	InitialAdminUsers []string
+}
+
+// Identity represents an authenticated user.
+type Identity interface {
+	// Allow reports whether the user should be allowed to access
+	// any of the users or groups in the given ACL slice.
+	Allow(ctx context.Context, acl []string) (bool, error)
+}
+
+// AdminACL holds the name of the administrator ACL.
+const AdminACL = "admin"
```

Comment:
I think I'd prefer this name to be a little more magic "_admin" would fit in with the other meta ACLs.

Reply:
I'm not sure. I *think* it's better for "admin" to be the root, because all other meta ACLs have something to be meta to, and I think it would be confusing if a service had two "admin" roles.
We could rename it to something else if you like. "acladmin" maybe. It's easy to change in a later PR before deployment.